code/headers/gga.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <hwlib.hpp>
+
+namespace r2d2::location {
+    /**
+     * @brief this struct contains essential fix data which provide 3D location
+     * and accuracy data
+     *
+     * https://www.gpsinformation.org/dale/nmea.htm#GGA
+     *
+     */
+    struct gga_s {
```

Comment:
It might be worth reordering members for better packing.

Reply:
I think the current ordering is okay as this is the order the data is arranged in the nmea message. This does use more memory but as we only have a single location at the time it shouldn't matter that much

Reply:
There was a bug on the sam3x8e with this struct. It somehow didn't place padding bytes or something as all the variables were not the same as when they were put in the struct. I changed the order and it worked again.